### ISSUE TITLE
Add a way to get the bounding box of any 3D Node including it's children (like shown in the editor)

### ISSUE THREAD

**Issue by TheContainer** [godotengine/godot-proposals#10032]
### Describe the project you are working on

A sandbox game, where you can spawn objects into the world.

### Describe the problem or limitation you are having in your project

I want to generate automatic images of the objects you can spawn in the spawn menu for automating purposes. However, based on the size of the objects, the camera taking these images will have to be at different distances so the camera doesn't clip. The best way I thought of would be to get the bounding box of the root node.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add a method to get the calculated bounding box. I know there is a get_aabb() function for VisualInstances, but what if you want to get the bounding box of a Node3D with multiple VisualInstances instead?

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Call the method to get the size of the bounding box as a Vector3.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Well, not that easy, but I found a [Reddit post](https://www.reddit.com/r/godot/comments/18bfn0n/how_to_calculate_node3d_bounding_box/), which also roughly describes the problem. It does contain multiple solutions, but it also has some decent lines of code.

### Is there a reason why this should be core and not an add-on in the asset library?

It's technically already implemented (because you can see the bounding box in the editor when selected). Although the use cases are probably not that high, the bounding box is pretty general.

**Comment by AThousandShips** [comment]
> Call the method to get the size of the bounding box as a Vector3.

The bounds should reasonably be an `AABB` I'd say

**Comment by TheContainer** [comment]
> > Call the method to get the size of the bounding box as a Vector3.
> 
> The bounds should reasonably be an `AABB` I'd say

True. Haven't thought that to the end. There is a size property in AABB.

**Comment by Straven35** [comment]
I found myself needing this functionality today, so I did some digging. The function that achieves this result in-editor is this one right here:

https://github.com/godotengine/godot/blob/master/editor/plugins/node_3d_editor_plugin.cpp#L4461

I copied the function over to GDscript to use myself, which can be found here:

```gdscript
func get_node_aabb(node : Node3D = null, ignore_top_level : bool = true, bounds_transform : Transform3D = Transform3D()) -> AABB:
	var box : AABB
	var transform : Transform3D

	# we are going down the child chain, we want the aabb of each subsequent node to be on the same axis as the parent
	if bounds_transform.is_equal_approx(Transform3D()):
		transform = node.global_transform
	else:
		transform = bounds_transform
	
	# no more nodes. return default aabb
	if node == null:
		return AABB(Vector3(-0.2, -0.2, -0.2), Vector3(0.4, 0.4, 0.4))
	# makes sure the transform we get isn't distorted
	var top_xform : Transform3D = transform.affine_inverse() * node.global_transform

	# convert the node into visualinstance3D to access get_aabb() function.
	var visual_result : VisualInstance3D = node as VisualInstance3D
	if visual_result != null:
		box = visual_result.get_aabb()
	else:
		box = AABB()
	
	# xforms the transform with the box aabb for proper alignment I believe?
	box = top_xform * box
	# recursion
	for i : int in thing.get_child_count():
		var child : Node3D = node.get_child(i) as Node3D
		if child && !(ignore_top_level && child.top_level):
			var child_box : AABB = get_node_aabb(child, ignore_top_level, transform)
			box = box.merge(child_box)
	
	return box
```

(To call the function I am doing `get_node_aabb(node as Node3D)`. I am not sure if the cast is important in this instance, or if godot handles the conversion when the function is called)

It would be nice to have this function available directly in the Node3D class though.

**Comment by SoapSpangledGames** [comment]
![Image](https://github.com/user-attachments/assets/81a93f63-fafd-4b71-a617-3a3440ba7c16)

What is ```thing```?

**Comment by solodeveloping** [comment]
~~It looks like this has be done for MultiMeshes too.~~

~~It would be nice if this was available too.~~

~~As far as I know, both the code here~~ (which is now unavailable), ~~and the code in the Reddit thread, are not handling MultiMeshes~~

=> I still don't know why MultiMeshes work but it's handled

<img width="1159" height="368" alt="Image" src="https://github.com/user-attachments/assets/e2c57aa2-76fa-42fd-a92d-97a57d3365d7" />

**Comment by solodeveloping** [comment]
I found the current implementation (permalink: https://github.com/godotengine/godot/blob/c96e11965f1568343c65d783833250a6a9411cb2/editor/scene/3d/node_3d_editor_plugin.cpp#L5525)

I don't get how this is handling MultiMeshes though.

**Comment by solodeveloping** [comment]
I don't get it actually.

The documentation says ([class_multimesh.html](https://docs.godotengine.org/en/stable/classes/class_multimesh.html)):

> Since instances may have any behavior, the AABB used for visibility must be provided by the user.

But when I call get_aabb() on the MultiMeshInstance3D, they seem fine (I haven't indicated anything):

<img width="710" height="396" alt="Image" src="https://github.com/user-attachments/assets/0e41da7b-c25a-4854-857c-7d8548358c57" />

(I'm using debug_draw_3D to draw the aabbs)
